### PR TITLE
add modification to the algorithm, and modified navigation after plan creation

### PR DESCRIPTION
**Pull request by kasaechao**
1. Modification is now included in the workout plan creation algorithm. 
2. a bit of refactoring to the filterExerciseQuery method. I removed the switch statement as it's not needed anymore. 
3. navigation to the next screen after plan creation is changed from 'navigation.replace' to 'navigation.goBack'

### CODE REVIEW

File: myFitnessTrainer/src/components/GenerateNewPlanAlgoButton.js [hunk 1]
```diff
@@ -112,15 +99,15 @@ const GenerateNewPlanAlgoButton = ({
...
+        navigation.goBack("Dashboard");
     };
 
     return (
         <TouchableOpacity
             style={styles.button}
             onPress={handleCreateNewPlanAlgo}
         >
-            <Text style={styles.buttonText}>Create Workout Plan</Text>
+            <Text style={styles.buttonText}>Generate Fitness Plan</Text>
```
Comment (ethorne2) [inline myFitnessTrainer/src/components/GenerateNewPlanAlgoButton.js]: I had updated this text to be 'Create Workout Plan', so we are consistent in referencing the plan as a 'workout plan' vs. 'fitness plan'. I'm good with changing it to 'Generate Workout Plan'. Thoughts?

**Review by ethorne2 — COMMENTED** [review]
small button text question then all else looks great!

File: myFitnessTrainer/src/components/GenerateNewPlanAlgoButton.js (hunk 1 above)
Comment (kasaechao) [reply myFitnessTrainer/src/components/GenerateNewPlanAlgoButton.js]: Sorry about the confusion. 'Create Workout Plan' sounds good.

**Review by ethorne2 — APPROVED** [review]
🚀

**Pull request merged**